client/blocks/plan-thank-you-card/index.jsx

```diff
@@ -0,0 +1,93 @@
+/**
+ * External dependencies
+ */
+import React, { Component, PropTypes } from 'react';
+import { localize } from 'i18n-calypso';
+import { connect } from 'react-redux';
+
+/**
+ * Internal dependencies
+ */
+import Gridicon from 'components/gridicon';
+import { getSite } from 'state/sites/selectors';
+import { getCurrentPlan } from 'state/sites/plans/selectors';
+import QuerySites from 'components/data/query-sites';
+import QuerySitePlans from 'components/data/query-site-plans';
+import { plansList } from 'lib/plans/constants';
+import formatCurrency from 'lib/format-currency';
+
+class PlanThankYouCard extends Component {
+	static propTypes = {
+		siteId: PropTypes.number.isRequired
+	};
+
+	render() {
+		const {
+			plan,
+			site,
+			translate,
+			siteId
+		} = this.props;
+		// Non standard gridicon sizes are used here because we use them as background pattern with various sizes and rotation
+		/* eslint-disable wpcalypso/jsx-gridicon-size */
+		return (
+			<div className="plan-thank-you-card">
+				<QuerySites siteId={ siteId } />
+				<QuerySitePlans siteId={ siteId } />
+				<div className="plan-thank-you-card__header">
+					<Gridicon className="plan-thank-you-card__main-icon" icon="checkmark-circle" size={ 140 } />
+					{ ! site
+						? <div className="plan-thank-you-card__plan-name is-placeholder"></div>
+						: <div className="plan-thank-you-card__plan-name">
+								{ translate( '%(planName)s Plan', {
+									args: { planName: plansList[ site.plan.product_slug ].getTitle() }
+								} ) }
+							</div>
+					}
+					{ ! plan
+						? <div className="plan-thank-you-card__plan-price is-placeholder"></div>
+						: <div className="plan-thank-you-card__plan-price">{ formatCurrency( plan.rawPrice, plan.currencyCode ) }</div>
```

Comment:
I don't think it's a problem. What I was saying is to use `getPlanRawPrice` selector for getting plan's price instead of directly looking it up on the `plan` object. As Kerry suggested, use selectors when available and add new ones if they don't exist. :)